pkg/testfixtures/storage/mysql.go

```diff
@@ -47,87 +46,32 @@ func (m *mySQLTestContainer) GetDatabaseSchemaVersion() int64 {
 func (m *mySQLTestContainer) RunMySQLTestContainer(t testing.TB) DatastoreTestContainer {
-	dockerClient, err := client.NewClientWithOpts(
-		client.FromEnv,
-		client.WithAPIVersionNegotiation(),
+	ctx := context.Background()
+
+	mysqlContainer, err := testcontainersmysql.RunContainer(ctx,
+		testcontainers.WithImage(mySQLImage),
+		testcontainers.WithEnv(map[string]string{
+			"MYSQL_DATABASE":      "defaultdb",
+			"MYSQL_ROOT_PASSWORD": "secret",
+		}),
+		testcontainers.WithWaitStrategy(wait.ForLog("port: 3306  MySQL Community Server - GPL")),
+		testcontainers.WithHostConfigModifier(func(hostConfig *container.HostConfig) {
+			hostConfig.Tmpfs = map[string]string{"/var/lib/mysql": ""}
+		}),
+		testcontainersmysql.WithDatabase("defaultdb"),
+		testcontainersmysql.WithUsername("root"),
+		testcontainersmysql.WithPassword("secret"),
 	)
 	require.NoError(t, err)
-	t.Cleanup(func() {
-		dockerClient.Close()
-	})
+	t.Cleanup(func() { require.NoError(t, mysqlContainer.Terminate(ctx)) })
 
-	allImages, err := dockerClient.ImageList(context.Background(), types.ImageListOptions{
-		All: true,
-	})
+	mysqlHost, err := mysqlContainer.Host(ctx)
 	require.NoError(t, err)
-
-	foundMysqlImage := false
-	for _, image := range allImages {
-		for _, tag := range image.RepoTags {
-			if strings.Contains(tag, mySQLImage) {
-				foundMysqlImage = true
-				break
-			}
-		}
-	}
-
-	if !foundMysqlImage {
-		t.Logf("Pulling image %s", mySQLImage)
-		reader, err := dockerClient.ImagePull(context.Background(), mySQLImage, types.ImagePullOptions{})
-		require.NoError(t, err)
-
-		_, err = io.Copy(io.Discard, reader) // consume the image pull output to make sure it's done
-		require.NoError(t, err)
-	}
-
-	containerCfg := container.Config{
-		Env: []string{
-			"MYSQL_DATABASE=defaultdb",
-			"MYSQL_ROOT_PASSWORD=secret",
-		},
-		ExposedPorts: nat.PortSet{
-			nat.Port("3306/tcp"): {},
-		},
-		Image: mySQLImage,
-	}
-
-	hostCfg := container.HostConfig{
-		AutoRemove:      true,
-		PublishAllPorts: true,
-		Tmpfs:           map[string]string{"/var/lib/mysql": ""},
-	}
-
-	name := fmt.Sprintf("mysql-%s", ulid.Make().String())
-
-	cont, err := dockerClient.ContainerCreate(context.Background(), &containerCfg, &hostCfg, nil, nil, name)
-	require.NoError(t, err, "failed to create mysql docker container")
-
-	t.Cleanup(func() {
-		t.Logf("stopping container %s", name)
-		timeoutSec := 5
-
-		err := dockerClient.ContainerStop(context.Background(), cont.ID, container.StopOptions{Timeout: &timeoutSec})
-		if err != nil && !client.IsErrNotFound(err) {
-			t.Logf("failed to stop mysql container: %v", err)
-		}
-		t.Logf("stopped container %s", name)
-	})
-
-	err = dockerClient.ContainerStart(context.Background(), cont.ID, container.StartOptions{})
-	if err != nil {
-		t.Fatalf("failed to start mysql container: %v", err)
-	}
-
-	containerJSON, err := dockerClient.ContainerInspect(context.Background(), cont.ID)
+	mysqlPort, err := mysqlContainer.MappedPort(ctx, "3306/tcp")
```

Comment:
You can leverage the `mysqlContainer.ConnectionString` to create the mysql connection, although I'm not sure how it fits in your current test structure for MySQL.

Reply:
it will need refactoring a bit, so I will consider in next PRs